test/integration/protocol_integration_test.cc

```diff
@@ -2297,4 +2297,29 @@ TEST_P(DownstreamProtocolIntegrationTest, Test100AndDisconnect) {
   EXPECT_EQ("503", response->headers().getStatusValue());
 }
 
+// Tests a filter that returns a FilterHeadersStatus::Continue after a local reply without
+// processing new metadata generated in decodeHeader
+TEST_P(DownstreamProtocolIntegrationTest, ContinueAfterLocalReplyWithMetadata) {
+  config_helper_.addFilter(R"EOF(
+  name: continue-after-local-reply-with-metadata-filter
+  typed_config:
+    "@type": type.googleapis.com/google.protobuf.Empty
+  )EOF");
+  initialize();
+
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  // Send a headers only request.
+  IntegrationStreamDecoderPtr response;
+  EXPECT_ENVOY_BUG(
```

Comment:
I think this test is the same as the `DownstreamProtocolIntegrationTest.ContinueAfterLocalReply` and it does not test the change you have made. To test you change the `decodeHeaders()` of your test filer should return `StopAll` and then the `decodeMetadata()` should just have `ASSERT(false);` in it. I think the exception should work too, but it would be harder to diagnose than ASSERT.
And the test should just test for response from local reply. If it hits the ASSERT in the `decodeMetadata` then it would fail.

Reply:
Thanks for review. Your are right, I have added StopIteration in decoderHeaders and removed the EXPECT_ENVOY_BUG. Aslo replace throw with assert.